subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/OutputScrapingExecutionFailure.java

```diff
@@ -319,9 +319,19 @@ public void assertHasCause(String message) {
             }
         }
 
+        @Override
+        public void assertHasFirstCause(String message) {
+            String firstCause = causes.stream().findFirst().orElse("");
+            if (!firstCause.equals(message)) {
+                throw new AssertionFailedError(String.format("Expected first cause '%s', got '%s'", message, firstCause));
```

Comment:
I don't mind this form, but I think you could replace the if with:
```suggestion
                assert firstCause.equals(message) : String.format("Expected first cause '%s', got '%s'", message, firstCause);
```

Reply:
I'm reluctant to use `assert` here as it requires runtime assertions to be enabled at runtime.